### PR TITLE
add 4.6 templates and Readme update for local vol file and block

### PR DESCRIPTION
**Pull request by Bhagyashreek8**

### CODE REVIEW

File: config-templates/redhat/local-volume-block/4.5/README.md [hunk 1]
```diff
@@ -102,6 +102,8 @@ ibmcloud sat storage assignment create --name localvol-block-assign --group satC
 **Example output**
 ![Image of output](https://github.com/Hemlata3011/ibm-satellite-storage/blob/master/config-templates/redhat/local-volume-block/4.5/localVolBlk.png)
 
+**Follow** the [link](https://docs.openshift.com/container-platform/4.6/storage/persistent_storage/persistent-storage-local.html) to create the persistent volume claim and attach the claim to a pod.
+
 ## Troubleshooting
 
 1. PVs are not getting created
```
Comment (derekpoindexter) [inline config-templates/redhat/local-volume-block/4.5/README.md:109]:
```suggestion
**Local PV creation fails**
```

File: config-templates/redhat/local-volume-block/4.5/README.md [hunk 2]
```diff
@@ -116,4 +118,43 @@ ibmcloud sat storage assignment create --name localvol-block-assign --group satC
 	Navigation Menu -> Satellite -> Clusters
    - Select the cluster and click on the overflow menu of that particular cluster.
    - Click on the Re-attach cluster and copy the displayed command and execute it.
```
Comment (derekpoindexter) [inline config-templates/redhat/local-volume-block/4.5/README.md:120]:
```suggestion
1. Navigate to your cluster in the [IBM Cloud console](https://cloud.ibm.com/satellite/clusters)
2. Select your cluster and click on the overflow menu.
3. Click **Re-attach** cluster.
4. Copy the reattach command and run it in your command line.
```

File: config-templates/redhat/local-volume-block/4.5/README.md [hunk 3]
```diff
@@ -116,4 +118,43 @@ ibmcloud sat storage assignment create --name localvol-block-assign --group satC
...
+   E0213 06:19:40.697628       1 diskmaker.go:203] failed to acquire lock on device /dev/xvde
+   E0213 06:19:40.697657       1 diskmaker.go:180] error symlinking /dev/xvdc to /mnt/local-storage/sat-local-block-gold/xvdc: error acquiring exclusive lock on /dev/xvdc
+   ```
+   delete the symlink from the node
+   ```
+   oc debug node/<node-name>
+   chroot /host
+   rm -rf </path/to/symlink/as/shown/in/logs>
+   ```
+   Example `rm -rf  /mnt/local-storage/sat-local-block-gold/xvdc`
```
Comment (derekpoindexter) [inline config-templates/redhat/local-volume-block/4.5/README.md:142]:
```suggestion
  5. Get the namespace where you deployed the local storage template and verify that the status is `Active`.
     ```sh
     oc get ns | grep local
     ```
     **Example output**
     ```
     local-storage                                      Active   101m
     ```
     
   6. Get the logs for the `local-disk-local-diskmaker` pod.
     ```sh
     kubectl logs -f pod/local-disk-local-diskmaker-7ww2j -n local-storage
     ```
     
     **Example output:**
     ```
     I0213 06:19:35.103830       1 diskmaker.go:24] Go Version: go1.13.15
     I0213 06:19:35.104141       1 diskmaker.go:25] Go OS/Arch: linux/amd64
     I0213 06:19:35.104148       1 diskmaker.go:26] local-storage-diskmaker Version: v4.5.0-202101300210.p0-0-ged6884f-dirty
     E0213 06:19:40.697628       1 diskmaker.go:203] failed to acquire lock on device /dev/xvde
     E0213 06:19:40.697657       1 diskmaker.go:180] error symlinking /dev/xvdc to /mnt/local-storage/sat-local-block-gold/xvdc: error acquiring exclusive lock on /dev/xvdc
     ```
     
  7. Log-in to the node where the diskmaker pod is deployed.
     ```sh
     oc debug node/<node-name>
     ```
   
   8. Run the following command to use host binaries.
      ```sh
      chroot /host
      ```
   9. Delete the symlink.
      ```sh
      rm -rf </path/to/symlink/as/shown/in/logs>
      ```
     
     **Example command:**
     ```sh
     rm -rf  /mnt/local-storage/sat-local-block-gold/xvdc
     ```
```
```

File: config-templates/redhat/local-volume-block/4.5/README.md [hunk 4]
```diff
@@ -116,4 +118,43 @@ ibmcloud sat storage assignment create --name localvol-block-assign --group satC
...
+
+2.  You can check the local-disk-local-provisioner pod logs to verify if the pv got created with specified volume
```
Comment (derekpoindexter) [inline config-templates/redhat/local-volume-block/4.5/README.md:144]:
```suggestion
10. Get the `local-disk-provisioner` pod logs and verify that PV creation is successful.
```

File: config-templates/redhat/local-volume-block/4.5/README.md [hunk 5]
```diff
@@ -116,4 +118,43 @@ ibmcloud sat storage assignment create --name localvol-block-assign --group satC
...
+    I0213 06:30:44.146148       1 common.go:382] Creating client using in-cluster config
+    I0213 06:30:44.174765       1 main.go:85] Starting controller
+    I0213 06:30:44.174800       1 main.go:100] Starting metrics server at :8080
+    I0213 06:30:44.175265       1 controller.go:45] Initializing volume cache
+    I0213 06:30:44.378164       1 controller.go:108] Controller started
+    I0213 06:30:44.378851       1 discovery.go:304] Found new volume at host path "/mnt/local-storage/sat-local-bolck-gold/xvdc" with capacity 53687091200, creating Local PV "local-pv-1d14680"
+    I0213 06:30:44.396328       1 cache.go:55] Added pv "local-pv-1d14680" to cache
+    I0213 06:30:44.396471       1 discovery.go:337] Created PV "local-pv-1d14680" for volume at "/mnt/local-storage/sat-local-block-gold/xvdc"
+    I0213 06:30:44.462244       1 cache.go:64] Updated pv "local-pv-1d14680" to cache
+    ```
```
Comment (derekpoindexter) [inline config-templates/redhat/local-volume-block/4.5/README.md:157]:
```suggestion
    ```sh
    kubectl logs -f pod/local-disk-local-provisioner-xstjh -n local-storage 
    ```
    
    **Example output:**
    ```
    ...
    I0213 06:30:44.146148       1 common.go:382] Creating client using in-cluster config
    I0213 06:30:44.174765       1 main.go:85] Starting controller
    I0213 06:30:44.174800       1 main.go:100] Starting metrics server at :8080
    I0213 06:30:44.175265       1 controller.go:45] Initializing volume cache
    I0213 06:30:44.378164       1 controller.go:108] Controller started
    I0213 06:30:44.378851       1 discovery.go:304] Found new volume at host path "/mnt/local-storage/sat-local-bolck-gold/xvdc" with capacity 53687091200, creating Local PV "local-pv-1d14680"
    I0213 06:30:44.396328       1 cache.go:55] Added pv "local-pv-1d14680" to cache
    I0213 06:30:44.396471       1 discovery.go:337] Created PV "local-pv-1d14680" for volume at "/mnt/local-storage/sat-local-block-gold/xvdc"
    I0213 06:30:44.462244       1 cache.go:64] Updated pv "local-pv-1d14680" to cache
    ```
```

**Review by derekpoindexter — APPROVED** [review]
left some commit suggestions - approved

**Review by derekpoindexter — APPROVED** [review]

**Comment by derekpoindexter** [comment]
^ see commit suggestions

**Review by nkkashyap — CHANGES_REQUESTED** [review]
Please post the test results for this PR.

**Comment by Bhagyashreek8** [comment]
Test results for local volume file -
```
# ibmcloud sat storage config create --name localvolfile-bha02 --template-name local-volume-file --template-version 4.5 -p "label-key=storage" -p "label-value=bhagyafile" -p "devicepath=/dev/xvdc" --source-org Bhagyashreek8 --source-branch develop
Creating Satellite storage configuration...
OK
Storage configuration 'localvolfile-bha02' was successfully created with ID 'e894017a-2462-4cdf-9874-0d14be822c91'.
root@bha-fyre-ubuntu161:~# 
root@bha-fyre-ubuntu161:~# 
root@bha-fyre-ubuntu161:~# ibmcloud sat storage assignment create --name test-file-bha02 --group sat-local  --config localvolfile-bha02
Creating assignment...
OK
Assignment test-file-bha02 was successfully created with ID ddcf7f28-9c4f-4fb1-9dd1-7cf8d8f7c324.
```

```
kubectl get nodes
NAME             STATUS   ROLES           AGE   VERSION
169.63.151.100   Ready    master,worker   13m   v1.18.3+e574db2
169.63.151.102   Ready    master,worker   11m   v1.18.3+e574db2
169.63.151.104   Ready    master,worker   12m   v1.18.3+e574db2
```

```
$ oc label nodes 169.63.151.104 "storage=bhagyafile"
node/169.63.151.104 labeled
```

**Comment by Bhagyashreek8** [comment]
```
$ oc get all -n local-storage -o wide
NAME                                         READY   STATUS    RESTARTS   AGE    IP               NODE             NOMINATED NODE   READINESS GATES
pod/local-disk-local-diskmaker-2qxwd         1/1     Running   0          95s    172.30.207.149   169.63.151.104   <none>           <none>
pod/local-disk-local-provisioner-m9vsc       1/1     Running   0          95s    172.30.207.148   169.63.151.104   <none>           <none>
pod/local-storage-operator-96c444dfc-qwdmg   1/1     Running   0          101s   172.30.140.221   169.63.151.102   <none>           <none>

NAME                             TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)     AGE   SELECTOR
service/local-storage-operator   ClusterIP   172.21.48.154   <none>        60000/TCP   98s   name=local-storage-operator

NAME                                          DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE   CONTAINERS                  IMAGES                                                                                                                                       SELECTOR
daemonset.apps/local-disk-local-diskmaker     1         1         1       1            1           <none>          97s   local-diskmaker             registry.redhat.io/openshift4/ose-local-storage-diskmaker@sha256:dc7a831d203e32afe4175b60d4dff75818f5322ff8dca5cb8d55ee37b16a7c0d            app=local-volume-diskmaker-local-disk
daemonset.apps/local-disk-local-provisioner   1         1         1       1            1           <none>          98s   local-storage-provisioner   registry.redhat.io/openshift4/ose-local-storage-static-provisioner@sha256:3560fa5f829a58ec71b09235817e90e4c75f3ead9037565bbbb3e82dbdc50fef   app=local-volume-provisioner-local-disk

NAME                                     READY   UP-TO-DATE   AVAILABLE   AGE    CONTAINERS               IMAGES                                                                                                                             SELECTOR
deployment.apps/local-storage-operator   1/1     1            1           104s   local-storage-operator   registry.redhat.io/openshift4/ose-local-storage-operator@sha256:80db9adda92c176880163f9f385b8d63d0a6ca4b4b8f253bf5957fe02140d73c   name=local-storage-operator

NAME                                               DESIRED   CURRENT   READY   AGE    CONTAINERS               IMAGES                                                                                                                             SELECTOR
replicaset.apps/local-storage-operator-96c444dfc   1         1         1       105s   local-storage-operator   registry.redhat.io/openshift4/ose-local-storage-operator@sha256:80db9adda92c176880163f9f385b8d63d0a6ca4b4b8f253bf5957fe02140d73c   name=local-storage-operator,pod-template-hash=96c444dfc



$ oc get sc | grep local
sat-local-file-gold       kubernetes.io/no-provisioner   Delete          WaitForFirstConsumer   false                  2m8s


$ oc get pv
NAME                CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM   STORAGECLASS          REASON   AGE
local-pv-dfe01aad   20Gi       RWO            Delete           Available           sat-local-file-gold            119s

```

**Comment by Bhagyashreek8** [comment]
```
$ cat pvc-file-devtest.yaml 
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: devtest-file-pvc
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 20Gi
  volumeMode: Filesystem
  storageClassName: sat-local-file-gold
```

```
$ cat pod-file-devtest.yaml 
apiVersion: v1
kind: Pod
metadata:
  name: pod-devtest-file-02
spec:
  volumes:
  - name: test-vol
    persistentVolumeClaim:
        claimName: devtest-file-pvc
  containers:
  - name: devtest-demo-container
    image: busybox
    command: ["sh"]
    args: ["-c", "while true; do echo $(date +%s) > /mnt/myvol/testfile; sleep 10; done"]
    volumeMounts:
    - name: test-vol
      mountPath: "/mnt/local"
```

```
$ kubectl create -f pvc-file-devtest.yaml 
persistentvolumeclaim/devtest-file-pvc created

$ kubectl create -f pod-file-devtest.yaml 
pod/pod-devtest-file-01 created
Bhagyashrees-MacBook-Pro:~ bhagyashree$ oc get pvc
NAME               STATUS   VOLUME              CAPACITY   ACCESS MODES   STORAGECLASS          AGE
devtest-file-pvc   Bound    local-pv-dfe01aad   20Gi       RWO            sat-local-file-gold   105s
Bhagyashrees-MacBook-Pro:~ bhagyashree$ 
Bhagyashrees-MacBook-Pro:~ bhagyashree$ 
Bhagyashrees-MacBook-Pro:~ bhagyashree$ 
Bhagyashrees-MacBook-Pro:~ bhagyashree$ oc get pod
NAME                  READY   STATUS    RESTARTS   AGE
pod-devtest-file-01   1/1     Running   0          20s
Bhagyashrees-MacBook-Pro:~ bhagyashree$ 
Bhagyashrees-MacBook-Pro:~ bhagyashree$ 
Bhagyashrees-MacBook-Pro:~ bhagyashree$ kubectl exec -it pod-devtest-file-01 /bin/sh
/ # cd /mnt/local/
/mnt/local # ls
lost+found
/mnt/local # cd lost\+found/
/mnt/local/lost+found # ls
/mnt/local/lost+found # cd ..
/mnt/local # vi test.txt
/mnt/local # exit
Bhagyashrees-MacBook-Pro:~ bhagyashree$ vi pod-file-devtest.yaml 
Bhagyashrees-MacBook-Pro:~ bhagyashree$ 
Bhagyashrees-MacBook-Pro:~ bhagyashree$ 
Bhagyashrees-MacBook-Pro:~ bhagyashree$ kubectl create -f pod-file-devtest.yaml 
pod/pod-devtest-file-02 created
Bhagyashrees-MacBook-Pro:~ bhagyashree$ kubectl get pods
NAME                  READY   STATUS    RESTARTS   AGE
pod-devtest-file-01   1/1     Running   0          2m9s
pod-devtest-file-02   1/1     Running   0          7s
Bhagyashrees-MacBook-Pro:~ bhagyashree$ kubectl exec -it pod-devtest-file-02 /bin/sh
/ # cd /mnt/local/
/mnt/local # ls
lost+found  test.txt
/mnt/local # exit
```

**Comment by Bhagyashreek8** [comment]
Test results for local volume block -
```
# oc label nodes 169.63.151.100 169.63.151.102 "storage=bhagya"
node "169.63.151.100" labeled
node "169.63.151.102" labeled
```

```
# ibmcloud sat storage config create --name localvolblock-bha01 --template-name local-volume-block --template-version 4.5 -p "label-key=storage" -p "label-value=bhagya" -p "devicepath=/dev/xvdc" --source-org Bhagyashreek8 --source-branch develop
Creating Satellite storage configuration...
OK
Storage configuration 'localvolblock-bha01' was successfully created with ID 'b133d00c-2caa-42cd-83e8-ad3826df1b8a'.
root@bha-fyre-ubuntu161:~# ibmcloud sat storage assignment create --name test-bha01 --group sat-local --config localvolblock-bha01
Creating assignment...
OK
Assignment test-bha01 was successfully created with ID 57b43901-a888-4ec4-872d-ab6e365a6cee.
```

**Comment by Bhagyashreek8** [comment]
```
$ oc get all -n local-storage
NAME                                         READY   STATUS    RESTARTS   AGE
pod/local-disk-local-diskmaker-958tz         1/1     Running   0          52s
pod/local-disk-local-diskmaker-bngnh         1/1     Running   0          52s
pod/local-disk-local-provisioner-cmhl5       1/1     Running   0          52s
pod/local-disk-local-provisioner-vzjcr       1/1     Running   0          52s
pod/local-storage-operator-96c444dfc-lgp7k   1/1     Running   0          5m32s

NAME                             TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)     AGE
service/local-storage-operator   ClusterIP   172.21.48.187   <none>        60000/TCP   5m25s

NAME                                          DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
daemonset.apps/local-disk-local-diskmaker     2         2         2       2            2           <none>          53s
daemonset.apps/local-disk-local-provisioner   2         2         2       2            2           <none>          53s

NAME                                     READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/local-storage-operator   1/1     1            1           5m34s

NAME                                               DESIRED   CURRENT   READY   AGE
replicaset.apps/local-storage-operator-96c444dfc   1         1         1       5m35s
Bhagyashrees-MacBook-Pro:~ bhagyashree$ 
Bhagyashrees-MacBook-Pro:~ bhagyashree$ 
Bhagyashrees-MacBook-Pro:~ bhagyashree$ oc get pv
NAME                CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM   STORAGECLASS           REASON   AGE
local-pv-306c726    20Gi       RWO            Delete           Available           sat-local-block-gold            60s
local-pv-95b0bdbc   20Gi       RWO            Delete           Available           sat-local-block-gold            64s

$ oc get sc | grep local
sat-local-block-gold      kubernetes.io/no-provisioner   Delete          WaitForFirstConsumer   false                  32s
```

**Comment by Bhagyashreek8** [comment]
```
# ibmcloud sat storage config ls
Listing Satellite storage configurations...
OK
Name                  ID                                     Template Name        Template Version   
localvolblock-bha01   b133d00c-2caa-42cd-83e8-ad3826df1b8a   local-volume-block   4.5   
localvolfile-bha02    e894017a-2462-4cdf-9874-0d14be822c91   local-volume-file    4.5   

# ibmcloud sat storage config get --config localvolblock-bha01
Getting Satellite storage configuration details...
OK
                       
Name:               localvolblock-bha01   
ID:                 b133d00c-2caa-42cd-83e8-ad3826df1b8a   
Template Name:      local-volume-block   
Template Version:   4.5   

Configuration Parameters
Name          Value   
devicepath    /dev/xvdc   
label-key     storage   
label-value   bhagya   


# ibmcloud sat storage config get --config localvolfile-bha02
Getting Satellite storage configuration details...
OK
                       
Name:               localvolfile-bha02   
ID:                 e894017a-2462-4cdf-9874-0d14be822c91   
Template Name:      local-volume-file   
Template Version:   4.5   

Configuration Parameters
Name          Value   
devicepath    /dev/xvdc   
fstype        ext4   
label-key     storage   
label-value   bhagyafile   
```

**Comment by Bhagyashreek8** [comment]
Test results on ocp 4.6 cluster for local volume file and block templates captured here -
https://github.ibm.com/alchemy-containers/armada-storage/issues/1868

**Comment by nkkashyap** [comment]
@Bhagyashreek8 
Need to add new templates to `/template_list.json`

**Review by nkkashyap — APPROVED** [review]
Looks good

**Review by nkkashyap — APPROVED** [review]
Looks good

**Pull request merged**